pkgs/tools/package-management/nix/config.nix

```diff
@@ -0,0 +1,69 @@
+{ runtimeShell, writeReferencesToFile, runCommand
+, nix, lib
+}:
+
+{ cfg }:
```

Comment:
If this is intended to be used outside of the module system it currently means that all config options must be defined, unlike with the module system there are no default values in this case.
Using instead `{ extraOptions ? '', maxJobs }:` etc. makes the arguments explicit and simplifies usage outside of nixos for options that make sense without a value.

Reply:
For non-NixOS users it's nicest to have the options here, along with defaults. At the same time, we shouldn't duplicate work. However, because this derivation would fail to build if certain attributes in `cfg` are missing, I am of the opinion they should be added directly as parameters of this function, and thus `cfg` be dropped.

Frankly, I don't mind so much how it looks. Either way works for me.

Reply:
I could of course (and I'd like this to be more common in nixpkgs) put the module options right next to the Nix package, such that anybody using nixpkgs can use the nice options and what comes with them (type checking, defaults, overridability, merging, etc.), the package would evaluate its own small module system. Then there wouldn't be any duplicated work either, the NixOS module can get the options from the Nix package (as a submodule). This would also work for any other package having a configuration associated with it but the options currently only being usable from NixOS.

Let me know what you think, if people agree I can do this.